x-pack/plugins/security/server/config_deprecations.ts

```diff
@@ -136,22 +136,25 @@ export const securityConfigDeprecationProvider: ConfigDeprecationProvider = ({
     }
   },
   (settings, fromPath, addDeprecation) => {
-    if (settings?.xpack?.security?.enabled === false) {
+    if ('enabled' in (settings?.xpack?.security || {})) {
       addDeprecation({
         title: i18n.translate('xpack.security.deprecations.enabledTitle', {
-          defaultMessage: 'Disabling the security plugin "xpack.security.enabled" is deprecated',
+          defaultMessage: 'Setting "xpack.security.enabled" is deprecated',
         }),
         message: i18n.translate('xpack.security.deprecations.enabledMessage', {
           defaultMessage:
-            'Disabling the security plugin "xpack.security.enabled" will only be supported by disable security in Elasticsearch.',
+            'Enabling or disabling the security plugin from Kibana using "xpack.security.enabled" is deprecated. This should instead be controlled via Elasticsearch.',
```

Comment:
```suggestion
            'Enabling or disabling the Security plugin in Kibana is deprecated. If you must, disable security in Elasticsearch.',
```

Reply:
I'm not sure about this change:

> If you must, disable security in Elasticsearch.

It only talks about the "disable" part, not enabling.

It's [recommended in the Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/security-minimal-setup.html#_enable_elasticsearch_security_features) to always set `xpack.security.enabled` to the desired value, no matter the default value (i.e. set it to `true` to make sure security is enabled or to `false` to make sure it's disabled) as the default value depends on whatever license you're currently running.

Reply:
I think that we should be specific as to where users must go to enable/disable security. For example:
> Enabling or disabling the Security plugin in Kibana is deprecated. To enable or disable security, set `xpack.security.enabled` to `true` or `false` in `elasticsearch.yml`. 

We could then point users to this URL, which indicates how to modify the value for that setting: https://www.elastic.co/guide/en/elasticsearch/reference/7.x/security-minimal-setup.html#_enable_elasticsearch_security_features

Reply:
I updated the text to your suggestion 👍 

However, we still need to find a solution to what we link to